aptos-move/e2e-move-tests/src/tests/scripts.rs

```diff
@@ -5,10 +5,118 @@ use crate::{assert_success, build_package, tests::common, MoveHarness};
+        .insert("example_addr".to_string(), *alice.address());
+
+    let result = h.publish_package_with_options(
+        &alice,
+        &common::test_dir_path("../../../move-examples/fungible_asset/managed_fungible_asset"),
+        build_options.clone(),
+    );
+
+    assert_success!(result);
+    let result = h.publish_package_with_options(
+        &alice,
+        &common::test_dir_path("../../../move-examples/fungible_asset/managed_fungible_token"),
+        build_options.clone(),
+    );
+    assert_success!(result);
+
+    assert_success!(h.run_entry_function(
+        &root,
+        str::parse(&format!(
+            "0x{}::coin::create_coin_conversion_map",
+            (*root.address()).to_hex()
+        ))
+        .unwrap(),
+        vec![],
+        vec![],
+    ));
+
+    let metadata = h
+        .execute_view_function(
+            str::parse(&format!(
+                "0x{}::managed_fungible_token::get_metadata",
+                (*alice.address()).to_hex()
+            ))
+            .unwrap(),
+            vec![],
+            vec![],
+        )
+        .values
+        .unwrap()
+        .pop()
+        .unwrap();
+    let metadata = bcs::from_bytes::<AccountAddress>(metadata.as_slice()).unwrap();
+
+    let result = h.run_entry_function(
+        &alice,
+        str::parse(&format!(
+            "0x{}::managed_fungible_asset::mint_to_primary_stores",
+            (*alice.address()).to_hex()
+        ))
+        .unwrap(),
+        vec![],
+        vec![
+            bcs::to_bytes(&metadata).unwrap(),
+            bcs::to_bytes(&vec![alice.address()]).unwrap(),
+            bcs::to_bytes(&vec![100u64]).unwrap(), // amount
+        ],
+    );
+    assert_success!(result);
+
+    let package = build_package(
+        common::test_dir_path("script_with_object_param.data/pack"),
+        build_options,
+    )
+    .expect("building package must succeed");
+
+    let code = package.extract_script_code().into_iter().next().unwrap();
+    let script = Script::new(code, vec![], vec![
+        TransactionArgument::Serialized(bcs::to_bytes(&metadata).unwrap()),
+        TransactionArgument::Serialized(bcs::to_bytes(&vec![alice.address()]).unwrap()),
+        TransactionArgument::Serialized(bcs::to_bytes(&vec![bob.address()]).unwrap()),
+        TransactionArgument::Serialized(bcs::to_bytes(&vec![30u64]).unwrap()),
+    ]);
+
+    let txn = TransactionBuilder::new(alice.clone())
+        .script(script.clone())
+        .sequence_number(13)
+        .max_gas_amount(1_000_000)
+        .gas_unit_price(1)
+        .sign();
+
+    let status = h.run(txn);
+    assert_success!(status);
+
+    h.enable_features(vec![], vec![FeatureFlag::ALLOW_SERIALIZED_SCRIPT_ARGS]);
+
+    let txn = TransactionBuilder::new(alice.clone())
+        .script(script.clone())
+        .sequence_number(14)
+        .max_gas_amount(1_000_000)
+        .gas_unit_price(1)
+        .sign();
+
+    let status = h.run(txn);
+    assert!(status.is_discarded());
```

Comment:
Hmmmm, this is a discard? Then we should really do this check somewhere early, or otherwise commit txn on chain if we keep it as is: right now we would do dependency traversal beforehand, so an attack vector can be to submit a script before the feature gets enabled and there are many dependencies?

Reply:
Ah good point. Lemme flip the two checks.

Reply:
Maybe even a new status code, @wrwg is adding one I think